File: BroadcastReceiver/src/main/java/com/example/broadcastReceiver/SendStandardBroadcastActivity.java
package com.example.broadcastReceiver;

import androidx.appcompat.app.AppCompatActivity;

import android.content.Intent;
import android.os.Bundle;

//发送自定义广播: 发送标准广播
/*注意：在Android8.0 系统之后，静态注册的BroadcastReceiver 是无法接收隐式广播的，
而默认情况下我们发出的自定义广播恰恰都是隐式广播。因此这里一定要调用setPackage()方法，指定这条广播是
发送给哪个应用程序的，从而让它变成一条显式广播，否则静态注册的BroadcastReceiver 将无法接收到这条广播*/
public class SendStandardBroadcastActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_send_standard_broadcast);

        findViewById(R.id.button).setOnClickListener(v -> {
            Intent intent = new Intent("com.example.broadcasttest.MY_BROADCAST");
            //传入当前应用程序的包名
            //getPackageName()的语法糖写法，用于获取当前应用程序的包名
            intent.setPackage(getPackageName());
            sendBroadcast(intent);
        });
    }
}

File: Kotlin/src/main/java/com/example/learnKotlin/chapter6/SendStandardBroadcastActivity.kt
package com.example.learnKotlin.chapter6

import android.content.Intent
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import com.example.Kotlin.R
import kotlinx.android.synthetic.main.activity_send_standard_broadcast.*

//发送标准广播
class SendStandardBroadcastActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_send_standard_broadcast)

        button.setOnClickListener{
            val intent = Intent("com.example.broadcasttest.MY_BROADCAST")
            //一定要调用setPackage()方法，指定这条广播是发送给哪个应用程序的，从而让它变成一条显式广播
            intent.setPackage(packageName)
            sendBroadcast(intent)
        }
    }
}